apps/admin/lib/API.php

```diff
@@ -24,4 +24,27 @@ public function post_toolbar () {
 		}
 		return Toolbar::save($out);
 	}
+	
+	public function get_async() {
+		if (!isset($_GET['page'])) return $this->error('No page route given.');
+		// parse any GET values from the original URL
+		$params = explode('?',$_GET['page']);
+		unset($_GET['page']);
+		$route = $params[0];
+		if(isset($params[1])) $params = explode('&',$params[1]);
+		foreach ($params as $param) {
+			list($key,$value) = explode('=',$param);
+			$_GET[$key] = $value;
+		}
+		// Set relevant URL globals that should contain the route value
+		$_SERVER['REQUEST_URI'] = $_COOKIE['elefant_last_page'] = $_GET['redirect'] = ($route == '/') ? '/index' : $route;
+		// This is sort of a hack to prevent any layouts from being rendered.
+		\Page::$bypass_layout = true; 
```

Comment:
That was the first thing I tried many ways before I added the ```Page::$bypass_layout```. Resources themselves can (and most likely will) set an alternate layout with ```$page->layout``` that just overrides that value, which in turn forces a standard return output of the HTML content with the layout. To combat this for the utility, I needed a global way to bypass all changes to just the layout value so it would return just the body content. Either that or add a new parameter to pass around with ```$controller->run``` to force layout to be false. That would be more work than my solution imo.